site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -1048,17 +1048,25 @@ public function saveSingleComponent() {
 
             $index = 0;
             // save existing marks
-            foreach ($component->getMarks() as $mark) {
+            foreach ($_POST['marks'] as $post_mark) {
                 if (isset($_POST['num_existing_marks'])) {
                     if ($index >= $_POST['num_existing_marks']) {
                         break;
                     }   
                 }
-                $mark->setPoints($_POST['marks'][$index]['points']);
-                $mark->setNote($_POST['marks'][$index]['note']);
-                $mark->setOrder($_POST['marks'][$index]['order']);
+                $mark = null;
+                foreach ($component->getMarks() as $cmark) {
+                    if ($cmark->getId() === $post_mark["id"]) {
+                        $mark = $cmark;
+                        break;
+                    }
+                }
+
+                $mark->setPoints($post_mark['points']);
+                $mark->setNote($post_mark['note']);
+                $mark->setOrder($post_mark['order']);
                 $mark->save();
-                $_POST['marks'][$index]['selected'] == 'true' ? $mark->setHasMark(true) : $mark->setHasMark(false);
+                $post_mark['selected'] == 'true' ? $mark->setHasMark(true) : $mark->setHasMark(false);
```

Comment:
`$mark->setHasMark($post_mark['selected'] == 'true');` or just `$mark->setHasMark($post_mark['selected'])`?

Reply:
Yeah that makes considerably more sense than this inline-if